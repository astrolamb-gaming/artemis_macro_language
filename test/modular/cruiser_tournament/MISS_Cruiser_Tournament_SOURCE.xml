<!-- Cruiser TOURNAMENT -->
<mission_data version="2.71" background_id_arme="" playerShipNames_arme="Artemis\Intrepid\Aegis\Horatio\Excalibur\Hera\Ceres\Diana">
  <imports>
    <import name="Cruiser_Tournament_MAP.xml" />
    <import name="Cruiser_Tournament_GAME.xml" />
    <import name="Cruiser_Tournament_Difficulty.xml" />
    <import name="Cruiser_Tournament_Mayday.xml" />
    <import name="Cruiser_Tournament_Anomolies.xml" />
    <import name="Cruiser_Tournament_Torgoth.xml" />
    <import name="Cruiser_Tournament_Kraliens.xml" />
    <import name="Cruiser_Tournament_Hunters.xml" />
    <import name="Cruiser_Tournament_Pirates.xml" />
    <import name="Cruiser_Tournament_Skaraan.xml" />
    <import name="Cruiser_Tournament_Stations.xml" />
    <import name="Cruiser_Tournament_BonusFleets.xml" />
    <import name="Cruiser_Tournament_Whales.xml" />
    <import name="Cruiser_Tournament_Cargo.xml" />
    <import name="Cruiser_Tournament_Destroyers.xml" />
    <!-- <import name="Cruiser_Tournament_periods.xml" /> -->
  </imports>
 <mission_description>
  A TSN Cruiser has 48 minutes to vanquish as many enemies as possible.^^For a full crew of 6 in Artemis 2.7.1.^^Your Comms Officer should take notes or print out the Communications Cheat Sheet in the mission folder.
  </mission_description>
  <start name="CRUISER TOURNAMENT">
    <!-- Players start in bottom right of of the 100,000 x 100,000 sector with one shuttle 'Pilgrim' aboard -->
    <set_player_carried_type player_slot="0" bay_slot="0" name="Dagger" raceKeys="TSN player" hullKeys="TSN Shuttle"/>
    <create type="player" player_slot="0" x="39055.0" y="0.0" z="85951.0" angle="295" name="Artemis" raceKeys="TSN player" hullKeys="Light Cruiser" warp="yes" jump="no"/>
    <set_object_property property="energy" value="1100" player_slot="0"/>
    <set_object_property property="countEMP" value="4" player_slot="0"/>
    <set_object_property property="countMine" value="6" player_slot="0"/>
    <set_object_property property="countNuke" value="2" player_slot="0"/>
    <set_object_property property="countHoming" value="8" player_slot="0"/>
    <set_object_property property="countPshock" value="2" player_slot="0"/>
    <!-- The Start Block also presents players with the mission title -->
    <big_message title="CRUISER TOURNAMENT" subtitle1="BY MIKE SUBSTELNY" subtitle2="a Challenge Tournament"/>
    <!-- Finally, the start block sets timers and variables to start the game -->
  </start>
</mission_data>