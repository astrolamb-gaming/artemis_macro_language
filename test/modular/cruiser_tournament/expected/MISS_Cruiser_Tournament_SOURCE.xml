<mission_data version="2.71" background_id_arme="" playerShipNames_arme="Artemis\Intrepid\Aegis\Horatio\Excalibur\Hera\Ceres\Diana">
  <mission_description>
  A TSN Cruiser has 48 minutes to vanquish as many enemies as possible.^^For a full crew of 6 in Artemis 2.7.1.^^Your Comms Officer should take notes or print out the Communications Cheat Sheet in the mission folder.
  </mission_description>
  <start name="CRUISER TOURNAMENT">
    
    <set_player_carried_type player_slot="0" bay_slot="0" name="Dagger" raceKeys="TSN player" hullKeys="TSN Shuttle"/>
    <create type="player" player_slot="0" x="39055.0" y="0.0" z="85951.0" angle="295" name="Artemis" raceKeys="TSN player" hullKeys="Light Cruiser" warp="yes" jump="no"/>
    <set_object_property property="energy" value="1100" player_slot="0"/>
    <set_object_property property="countEMP" value="4" player_slot="0"/>
    <set_object_property property="countMine" value="6" player_slot="0"/>
    <set_object_property property="countNuke" value="2" player_slot="0"/>
    <set_object_property property="countHoming" value="8" player_slot="0"/>
    <set_object_property property="countPshock" value="2" player_slot="0"/>
    
    <big_message title="CRUISER TOURNAMENT" subtitle1="BY MIKE SUBSTELNY" subtitle2="a Challenge Tournament"/>
    
    <set_object_property property="nebulaIsOpaque" value="0"/>
    <set_object_property property="sensorSettingspec" value="0"/>
    <create count="35" type="asteroids" startAngle="100" endAngle="300" startX="50000.0" startY="0.0" startZ="25000.0" radius="22000" randomRange="1200" randomSeed="3"/>
    <create count="35" type="asteroids" startAngle="20" endAngle="110" startX="70000.0" startY="0.0" startZ="40000.0" radius="23000" randomRange="1000" randomSeed="4"/>
    <create count="45" type="asteroids" startAngle="200" endAngle="300" startX="30000.0" startY="0.0" startZ="65000.0" radius="33000" randomRange="5000" randomSeed="2"/>
    <create count="35" type="nebulas" startAngle="200" endAngle="300" startX="30000.0" startY="0.0" startZ="65000.0" radius="33000" randomRange="5000" randomSeed="2"/>
    <create count="25" type="nebulas" startAngle="100" endAngle="300" startX="50000.0" startY="0.0" startZ="25000.0" radius="22000" randomRange="3200" randomSeed="3"/>
    <create count="25" type="nebulas" startAngle="20" endAngle="110" startX="70000.0" startY="0.0" startZ="40000.0" radius="23000" randomRange="3000" randomSeed="4"/>
    
    <set_skybox_index index="27"/>
    <log text="CRUISER TOURNAMENT has started"/>
    <log text="Beginning First Period. 48 minutes remaining."/>
    <set_timer name="credits_timer1" seconds="10"/>
    <set_timer name="Eight_Minute_timer" seconds="480"/>
    <set_timer name="One_Minute_timer" seconds="60"/>
    <set_variable name="Tonnage" value="0" integer="yes"/>
    <set_variable name="Period_Counter" value="0" integer="yes"/>
    <set_variable name="Minutes" value="48" integer="yes"/>
    <set_variable name="Difficult_Factor" value="0" integer="yes"/>
    <set_variable name="Bonus" value="0" integer="yes"/>
    <set_comms_button text="Request Score Report" sideValue="2" player_slot="0"/>
  </start>
  <event name="Time is up">
    <if_timer_finished name="endint_timer"/>
    <if_variable name="terminate" comparator="EQUALS" value="0"/>
    <if_variable name="Period_Counter" comparator="EQUALS" value="6"/>
    <big_message title="FINAL SCORE |tonnage| kilotons" subtitle1="Your detailed battle record" subtitle2="is in the mission log file."/>
    <set_variable name="terminate" value="1"/>
    <set_timer name="endint_timer" seconds="8"/>
  </event>
  <event name="Defeat">
    <if_not_exists name="Artemis"/>
    <if_variable name="terminate" comparator="NOT" value="1"/>
    <set_skybox_index index="17"/>
    <create type="player" player_slot="0" x="9555" y="0" z="85951" angle="295" name="Artemis" accent_color="99" raceKeys="TSN player" hullKeys="Scout" warp="yes"/>
    <big_message title="YOUR SHIP WAS DESTROYED" subtitle1="You scored |tonnage| kt of enemy ships." subtitle2="Your battle record is in the log file."/>
    <set_timer name="endint_timer" seconds="15"/>
    <set_variable name="terminate" value="1"/>
  </event>
  <event name="Game Over">
    <if_variable name="terminate" comparator="EQUALS" value="1"/>
    <if_variable name="gameOver" comparator="NOT" value="1"/>
    <if_timer_finished name="endint_timer"/>
    <end_mission/>
    <set_variable name="gameOver" value="1"/>
  </event>
  <event name="Credits 1">
    <if_timer_finished name="credits_timer1"/>
    <if_variable name="credits1" comparator="EQUALS" value="0"/>
    <big_message title="Written for TSN Light Cruiser" subtitle1="with an cadette crew" subtitle2="difficulty chosen by you"/>
    <set_variable name="credits1" value="1"/>
    <set_object_property property="topSpeed" value="1.2" name="Phoenix"/>
    <set_object_property property="turnRate" value="1/100" name="Phoenix"/>
  </event>
  <event name="Report Score">
    <if_timer_finished name="report_timer"/>
    <if_variable name="report_score" comparator="EQUALS" value="1"/>
    
    <big_message title="Current score = |Tonnage| kilotons" subtitle1="Time = |Minutes| minutes remaining" subtitle2=""/>
    <set_variable name="report_score" value="0"/>
    <log text="Score = |Tonnage| kt  |Minutes| minutes remaining."/>
  </event>
  <event name="Score Message">
    <if_comms_button text="Request Score Report"/>
    <incoming_comms_text from="Scorekeeper">
      Current score = |Tonnage| kilotons with |Minutes| minutes remaining
    </incoming_comms_text>
  </event>
  <event name="Count Minutes">
    <if_timer_finished name="One_Minute_timer"/>
    <set_variable name="Minutes" value="Minutes-1"/>
    <set_timer name="One_Minute_timer" seconds="60"/>
    <clear_comms_button text="Request Score Report"/>
    <clear_comms_button text="Request Cargo Report"/>
    <set_comms_button text="Request Score Report" sideValue="2" player_slot="0"/>
    <set_comms_button text="Request Cargo Report" sideValue="2" player_slot="0"/>
    
  </event>
</mission_data>